src/App.tsx

```diff
@@ -15,71 +15,100 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+export const App: React.FC = () => {
+  const defaultGood = 'Jam';
+  const defaultSelection = goods.indexOf(defaultGood);
+  const [selectedGood, setSelectedGood] = useState(defaultGood);
+  const [
+    selectedListItem,
+    setSelectedListItem,
+  ] = useState<string | null>(defaultSelection.toString());
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  const AddSelectHandler = (
+    event: React.MouseEvent,
+    good: string,
+  ) => {
+    const currTarget = event.currentTarget.closest('tr');
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+    if (currTarget) {
+      const selectedItem = currTarget.dataset.selected || null;
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+      setSelectedListItem(selectedItem);
+    }
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+    setSelectedGood(good);
+  };
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  const clearHandler = () => {
+    setSelectedListItem(null);
+    setSelectedGood('');
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {selectedGood ? (
+          `${selectedGood} is selected`
+        ) : (
+          'No goods selected'
+        )}
```

Comment:
make sure that you don't have complex structures in the return statement
you can create a variable before the return and simply use it